src/v/archival/ntp_archiver_service.h

```diff
@@ -655,6 +709,14 @@ class ntp_archiver {
     // be triggered by notify_leadership.
     ss::condition_variable _leader_cond;
 
+    // Used by clients wait()'ing on flush. Will be signalled by a complete
+    // flush(), or by a change in leadership.
+    ss::condition_variable _flush_cond;
+
+    // Indicates that a request to flush all local data up to offset() has been
+    // made when has_value() is true.
+    std::optional<model::offset> _flush_uploads_offset;
```

Comment:
+1 i'm pro std::optional<offset> rather than relying on the offset::min() pattern.